cas-server-support-osf/src/test/java/io/cos/cas/authentication/handler/support/OpenScienceFrameworkPrincipalFromRequestRemoteUserNonInteractiveCredentialsActionTests.java

```diff
@@ -71,11 +94,31 @@ public void handleInstitutionMissingInstitutionId() throws Exception {
 
         final OpenScienceFrameworkCredential osfCredential = new OpenScienceFrameworkCredential();
         osfCredential.setUsername(AbstractTestUtils.CONST_MAIL);
-        osfCredential.setInstitutionId("");
+        osfCredential.setInstitutionId(AbstractTestUtils.CONST_INSTITUTION_ID);
+        try {
+            osfRemoteAuthenticate.notifyRemotePrincipalAuthenticated(osfCredential);
+        } catch (final AccountException e) {
+            assertEquals(e.getMessage(), "Invalid remote principal: missing names.");
+            throw e;
+        }
+    }
+
+    @Test (expected = RemoteUserFailedLoginException.class)
+    public void handleInstitutionValidRemotePrincipal() throws Exception {
```

Comment:
Added a test for the case where all required attributes are provided and non-empty.